src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableBasedSchemaCache.java

```diff
@@ -84,6 +86,10 @@ protected Uni<SchemaObject> getSchemaObject(
     }
   }
 
+  Optional<SchemaObject> peekSchemaObject(String tableName) {
+    return Optional.ofNullable(schemaObjectCache.getIfPresent(tableName));
+  }
+
   private Uni<SchemaObject> loadSchemaObject(RequestContext requestContext, String collectionName) {
```

Comment:
Should change "collection" references in names to "table" since at this level they are tables of different types, not just collections.

Reply:
I was trying  to leave this class as much as possible , will cleanup in https://github.com/stargate/data-api/issues/2070

Reply:
makes sense.